core/iwasm/common/wasm_runtime_common.c

```diff
@@ -7739,3 +7748,114 @@ wasm_runtime_is_underlying_binary_freeable(WASMModuleCommon *const module)
 
     return true;
 }
+
+/*TODO: take us(below) out when have a linker */
+void
+wasm_runtime_destroy_imports(WASMExternInstance *extern_inst)
+{
+    if (extern_inst == NULL)
+        return;
+
```

Comment:
Seems we should also traverse the imports in the extern_inst, and if the import is memory instance, we should also free it? See below code in wasm_runtime_create_imports: `extern_instance->u.memory = wasm_runtime_create_memory`